src/logic/safe/utils/modules.ts

```diff
@@ -0,0 +1,74 @@
+import semverLessThan from 'semver/functions/lt'
+
+import { getGnosisSafeInstanceAt, SENTINEL_ADDRESS } from 'src/logic/contracts/safeContracts'
+import { ModulePair } from 'src/logic/safe/store/models/safe'
+import { SafeInfo } from 'src/logic/safe/utils/safeInformation'
+
+type ModulesPaginated = {
+  array: string[]
+  next: string
+}
+
+const buildModulesLinkedList = (modules: string[], nextModule: string = SENTINEL_ADDRESS): Array<ModulePair> | null => {
+  if (modules?.length) {
+    return modules.map((moduleAddress, index, modules) => {
+      const prevModule = modules[index + 1]
+      return [moduleAddress, prevModule !== undefined ? prevModule : nextModule]
+    })
+  }
+
+  // no modules
+  return null
+}
+
+/**
+ * Returns a list of Modules if there's any, in the form of [module, prevModule][]
+ * so we have an easy track of the prevModule and the currentModule when calling `disableModule`
+ *
+ * There's a slight difference on how many modules `getModules` return, depending on the Safe's version we're in:
+ *  - for >= v1.1.1 it will return a list of up to 10 modules
+ *  - for previous version it will return a list of all the modules enabled
+ *
+ * As we're using the safe-transactions service, and it's querying `getModules`,
+ *   we'll fallback to `getModulesPaginated` RPC call when needed.
+ *
+ * @todo: Implement pagination for `getModulesPaginated`. We're passing an arbitrary large number to avoid pagination.
+ *
+ * @param {SafeInfo | undefined } safeInfo
+ * @returns Array<ModulePair> | null | undefined
+ */
+export const getModules = async (safeInfo: SafeInfo | void): Promise<Array<ModulePair> | null | undefined> => {
+  if (safeInfo) {
```

Comment:
can you inverse the condition? `if (!safeInfo)` that will remove 1 nested level

Reply:
done this